### ISSUE TITLE
OptimInstance$eval_batch fails with empty input

### ISSUE THREAD

**Issue by mb706** [mlr-org/bbotk#129]
```r
> oi <- OptimInstanceSingleCrit$new(
    ObjectiveRFun$new(function(xs) xs,
    ps(x = p_dbl()),
    ps(x = p_dbl(tags = "maximize"))), terminator = trm("none")
  )
> oi$eval_batch(data.table(x = numeric(0)))
Error in setnames(x, value) : 
  Can't assign 1 names to a 0 column data.table
```
This could also be because of Objective, ObjectiveRFun or some other thing on the way. Should probably be tested.

**Comment by be-marc** [comment]
Fixed by 082860304270061165fac19dae1ba65ca437ed16.

**Comment by mb706** [comment]
Should we not allow `eval_batch()` with zero rows (and instead prevent the error from being thrown)?
